modules/ROOT/pages/sysconfig-network-configuration.adoc

```diff
@@ -0,0 +1,978 @@
+= Host Network Configuration
+
+== Host Network Configuration Options
+
+=== Background
+
+By default, Fedora CoreOS (FCOS) will attempt DHCP on every interface
+with a cable plugged in. However, if you need to use static addressing or more
+complex networking (vlans, bonds, bridges, teams, etc..), you can do
+so in a number of ways which are summarized below. Regardless of the
+way you choose to configure networking it all ends up as configuration
+for NetworkManager, which takes the form of NetworkManager keyfiles.
+More information on the keyfile format can be found
+https://developer.gnome.org/NetworkManager/stable/nm-settings-keyfile.html[here].
+More information on the subsection options for keyfiles can be found
+https://developer.gnome.org/NetworkManager/stable/ref-settings.html[here].
+
+=== Configuration Options
+
+FCOS machines are primarily configured via Ignition, which runs from the initramfs
+on the machine's first boot. Depending on the platform the machine may need
+network access to retrieve remote resources; either the Ignition config itself,
+or remote resources specified inside the Ignition config.
+
+NOTE: Networking will only be started in the initramfs if determined
+      to be required, or if explicitly requested by the user with
+      `*rd.neednet=1*`.
+
+Whether or not a machine needs networking in the initramfs can dictate how a
+user will configure networking for the machine. The options for
+configuring networking for a machine are:
+
+* via kernel arguments
+** these get processed by dracut modules in the initramfs on first boot
+* via `coreos-installer --copy-network`
+** by propagating the ISO/PXE install environment networking configuration 
+* via Afterburn
+** by applying network configuration injected by various platforms
+* via Ignition
+** by laying down files that NetworkManager then uses on startup
+
+NOTE: If you need networking connectivity to pull your Ignition
+      configuration, or if your Ignition has remote references, you
+      won't be able to provide your networking configuration via
+      Ignition.
+
+NOTE: If you provide networking configuration in multiple ways (i.e.
+      via kernel arguments and via Ignition) then the configuration
+      supplied via Ignition will win and be what is applied to the
```

Comment:
Worth noting here that `--copy-network` does win over kernel args? Meh, OK not mentioning it either. I think the general spirit of not specifying multiple times is correct.

Reply:
I'll leave it like it is I think